include/swift/Remote/MetadataReader.h

```diff
@@ -745,15 +813,20 @@ class MetadataReader {
                                  &FlaggedArgumentAddress))
           return BuiltType();
 
+        FunctionParam<BuiltType> Param;
+
         // TODO: Use target-agnostic FlaggedPointer to mask this!
         const auto InOutMask = (StoredPointer) 1;
-        ArgumentIsInOut.push_back((FlaggedArgumentAddress & InOutMask) != 0);
-        FlaggedArgumentAddress &= ~InOutMask;
+        if ((FlaggedArgumentAddress & InOutMask) != 0)
+          Param.setInOut();
 
-        if (auto ArgumentTypeRef = readTypeFromMetadata(FlaggedArgumentAddress))
-          Arguments.push_back(ArgumentTypeRef);
-        else
+        FlaggedArgumentAddress &= ~InOutMask;
```

Comment:
`shared` and `variadic`, too? It's odd to have them partially-supported, so I suggest either completing support or leaving a FIXME to do it all later.

Reply:
Yes, that's the next step after `MetadataReader` is updated to support these flags I'm going to update function metadata itself to include all of the flags for parameters...

Reply:
Added a FIXME about it on L807, going to address after this changes are in.